src/Engine/ProtoCore/FFI/CLRObjectMarshler.cs

```diff
@@ -738,10 +738,6 @@ public override object UnMarshal(StackValue dsObject, ProtoCore.Runtime.Context
         /// <returns>FFIObjectMarshler or null</returns>
         private FFIObjectMarshler GetMarshalerForCLRType(Type clrType, AddressType dsType)
```

Comment:
Please update the comment, to say that "even if dsType is Pointer, if clrType is collection then we marshal the ds object as a collection".